### PR TITLE
A bunch of fixes & improvements

### PR DESCRIPTION
**Pull request by zeenix**

### CODE REVIEW

**Comment by zeenix** [comment]
Converting to draft. Caught an issue while testing..

File: src/config/mod.rs [hunk 1]
```diff
@@ -329,10 +329,9 @@ impl<'c> Builder<'c> {
         self
     }
 
-    /// Set the jailer process configuration.
-    pub fn jailer_cfg(mut self, jailer_cfg: Option<Jailer<'c>>) -> Self {
-        self.0.jailer_cfg = jailer_cfg;
-        self
+    /// Create the jailer process configuration builder.
+    pub fn jailer_cfg(self) -> JailerBuilder<'c> {
```
Comment (loony-bean) [inline src/config/mod.rs:337]: not sure if we should rename the method to have "builder" in the name

File: src/config/mod.rs [hunk 2]
```diff
@@ -395,19 +394,17 @@ mod tests {
...
-
         let root_drive = Drive::builder("root", Path::new("/tmp/debian.ext4"))
             .is_root_device(true)
             .build();
 
         let config = Config::builder(Path::new("/tmp/kernel.path"))
             .vm_id(id)
-            .jailer_cfg(Some(jailer))
+            .jailer_cfg()
+            .chroot_base_dir(Path::new("/chroot"))
```
Comment (loony-bean) [inline src/config/mod.rs:377]: some indent issues?

File: src/config/mod.rs [hunk 3]
```diff
@@ -336,15 +338,6 @@ impl<'c> Builder<'c> {
         JailerBuilder::new(self)
     }
 
-    /// Set a unique identifier for this VM.
-    ///
-    /// It's set to a random uuid if not provided by the user. It's used as the Firecracker's
-    /// instance ID.
-    pub fn vm_id(mut self, vm_id: Uuid) -> Self {
```
Comment (loony-bean) [inline src/config/mod.rs:332]: not sure why we remove that?

Comment (loony-bean) [reply src/config/mod.rs:332]: oh it's a setter I see now

File: src/config/mod.rs (hunk 2 above)
Comment (loony-bean) [reply src/config/mod.rs:377]: yeah

**Review by loony-bean — APPROVED** [review]

File: src/config/mod.rs (hunk 1 above)
Comment (zeenix) [reply src/config/mod.rs:337]: the more I think about this, the more I think `_builer` suffix won't bring anything. So I'll go w/o it for now and let's think about renaming later if you disagree?

**Pull request merged**